pydantic_ai_slim/pydantic_ai/models/mistral.py

```diff
@@ -639,7 +639,8 @@ async def _get_event_iterator(self) -> AsyncIterator[ModelResponseStreamEvent]:
             content = choice.delta.content
             text, thinking = _map_content(content)
             for thought in thinking:
-                self._parts_manager.handle_thinking_delta(vendor_part_id='thinking', content=thought)
+                for event in self._parts_manager.handle_thinking_delta(vendor_part_id='thinking', content=thought):
+                    yield event
```

Comment:
not sure if this was a bug or this call was meant so simply update the parts manager, do you know @DouweM ?